pyluna-pathology/luna/pathology/cli/dsa_annotation_etl.py

```diff
@@ -0,0 +1,238 @@
+# General imports
+import os, json, logging, yaml
+import click
+
+from luna.common.custom_logger   import init_logger
+
+init_logger()
+logger = logging.getLogger("dsa_annotation_etl")
+
+from luna.common.utils import cli_runner
+
+_params_ = [('input_dsa_endpoint', str), ('collection_name', str), ('annotation_name', str), ('num_cores', int), ('username', str), ('password', str), ('output_dir', str)]
+
+@click.command()
+@click.argument('input_dsa_endpoint', nargs=1)
+@click.option('-o', '--output_dir', required=False,
+              help='path to output directory to save results')
+@click.option('-c', '--collection-name', required=False,
+              help='name of the collection to pull data from in DSA')
+@click.option('-a', '--annotation-name', required=False,
+              help='name of the annotations to pull from DSA (same annotation name for all slides)')
+@click.option('-u', '--username', required=False,
+              help='DSA username, can be inferred from DSA_USERNAME')
+@click.option('-p', '--password', required=False,
+              help='DSA password, should be inferred from DSA_PASSWORD')
+@click.option('-nc', '--num_cores', required=False,
+              help='Number of cores to use', default=4)
+@click.option('-m', '--method_param_path', required=False,
+              help='path to a metadata json/yaml file with method parameters to reproduce results')
+def cli(**cli_kwargs):
+    """ A cli tool
+
+    \b
+    Inputs:
+        input_dsa_endpoint: Path to the DSA endpoint like http://localhost:8080/dsa/api/v1
+    \b
+    Outputs:
+        slide_annotation_dataset
+    \b
+    Example:
+        export DSA_USERNAME=username
+        export DSA_PASSWORD=password
+        dsa_annotation_etl http://localhost:8080/dsa/api/v1 
+            --collection-name tcga-data
+            --annotation-name TumorVsOther
+            -o /data/annotations/
+    """
+    cli_runner( cli_kwargs, _params_, dsa_annotation_etl)
+
+import girder_client
+import pandas as pd
+from pathlib import Path
+
+from dask.distributed import Client, as_completed
+from tqdm import tqdm 
+
+### Transform imports 
+def dsa_annotation_etl(input_dsa_endpoint, username, password, collection_name, annotation_name, num_cores, output_dir):
+    """ Take
+
+    Args:
+        input_dsa_endpoint (str): path to input data
+        output_dir (str): output/working directory
+
+    Returns:
+        dict: metadata about function call
+    """
+    girder = girder_client.GirderClient(apiUrl=input_dsa_endpoint)
+
+    # Initial connnection
+    try:
+        girder.authenticate(username, password)
+        logger.info(f"Connected to DSA @ {input_dsa_endpoint}")
+    except Exception as exc:
+        logger.error(f"Couldn't authenticate DSA: {exc}")
+        raise RuntimeError("Connection to DSA endpoint failed.")
+
```

Comment:
See get_collection_uuid and get_item_uuid https://github.com/msk-mind/luna/blob/c638f55a7d544dc29ca80797e2c0801ddf0079d7/pyluna-pathology/luna/pathology/cli/dsa/dsa_api_handler.py#L10